content/chainguard/chainguard-images/reference/php/getting-started.md

```diff
@@ -0,0 +1,220 @@
+nano namegen
+```
+
+The following PHP script defines a minimalist CLI app with a single command called "get". It returns a random name based on a list of nouns and a list of adjectives.
+
+```php
+#!/usr/bin/php
+<?php
+
+require __DIR__ . '/vendor/autoload.php';
+
+use Minicli\App;
+
+$app = new App();
+
+$app->registerCommand('get', function () use ($app) {
+
+    $animals = [ 'turtle', 'seagull', 'octopus', 'shark', 'whale', 'dolphin', 'walrus', 'penguin', 'seahorse'];
+    $adjectives = [ 'ludicrous', 'mischievous', 'graceful', 'fortuitous', 'charming', 'ravishing', 'gregarious'];
+
+    $app->getPrinter()->info($adjectives[array_rand($adjectives)] . '-' . $animals[array_rand($animals)]);
+});
+
+$app->runCommand($argv);
+
+```
+
+Copy this content to your `namegen` script, save and close the file. Next, make the script executable with:
+
+```shell
+chmod +x namegen
+```
+
+You can now execute the app to test it out. If you have `php` locally, run:
+
+```shell
+php namegen get
+```
+
+If you're using the Docker setup, run:
+
+```shell
+docker run --rm -v ${PWD}:/work \
+    cgr.dev/chainguard/php:latest-glibc \
+    /work/namegen get
+```
+
+The command should output a random name combination:
+
+```shell
+ludicrous-walrus
+```
+
+The demo application is now ready. In the next step, you'll create a Dockerfile to run your app.
+
+## Step 2: Creating the Dockerfile
+
+To make sure our final image is _distroless_ while still being able to install dependencies with Composer, our build will consist of **two** stages: first, we'll build the application using the `glibc-dev` image variant, which is Wolfi-based and includes Composer and other useful tools for development.
+Then, we'll create a separate stage for the final image, which will be based on the `glibc` variant. That is a distroless image also based on Wolfi, which means it doesn't come with Composer or even a shell.
+
+Create a Dockerfile with:
+
+```shell
+touch Dockerfile
+```
+
+Then open this file in your code editor of choice, for example `nano`:
+
+```shell
+nano Dockerfile
+```
+The following Dockerfile will:
+
+1. Start a new build stage based on the `php:latest-glibc-dev` image and call it `builder`;
+2. Copy files from the current directory to the `/app` location in the container;
+3. Enter the `/app` directory and run `composer install` to install any dependencies;
+4. Start a new build stage based on the `php:latest-glibc`;
+5. Copy the application from the `builder` stage;
+6. Set up the application as entry point for this image.
+
+Copy this content to your own `Dockerfile`:
+
+```Dockerfile
+FROM cgr.dev/chainguard/php:latest-glibc-dev AS builder
+COPY guides /app
+RUN cd /app && \
+    composer install --no-progress --no-dev --prefer-dist
+
+FROM cgr.dev/chainguard/php:latest-glibc
+COPY --from=builder /app /app
+
+ENTRYPOINT [ "php", "/app/namegen" ]
+```
+Save the file when you're finished.
+
+You can now build the image with:
+
+```shell
+docker build . -t php-namegen
```

Comment:
I am still getting an issue with this command

On Debian:

```
Sending build context to Docker daemon  296.4kB
Step 1/6 : FROM cgr.dev/chainguard/php:latest-glibc-dev AS builder
 ---> f06865c7bb16
Step 2/6 : COPY guides /app
COPY failed: file not found in build context or excluded by .dockerignore: stat guides: file does not exist
```

on macOS:

```
[+] Building 0.1s (8/10)                                                        
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 303B                                       0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => [internal] load metadata for cgr.dev/chainguard/php:latest-glibc       0.0s
 => [internal] load metadata for cgr.dev/chainguard/php:latest-glibc-dev   0.0s
 => [internal] load build context                                          0.0s
 => => transferring context: 2B                                            0.0s
 => CACHED [builder 1/3] FROM cgr.dev/chainguard/php:latest-glibc-dev      0.0s
 => CACHED [stage-1 1/2] FROM cgr.dev/chainguard/php:latest-glibc          0.0s
 => ERROR [builder 2/3] COPY guides /app                                   0.0s
------
 > [builder 2/3] COPY guides /app:
------
failed to compute cache key: "/guides" not found: not found
```

Reply:
oh I see the problem, it is really strange I have no idea how this "guides" showed up in the Dockerfile, it was probably a faulty find and replace. Sorry! I just updated it now.